### PR TITLE
Short string optimization example for C.180

### PR DESCRIPTION
**Pull request by Eliyahu-Ravuna**

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -7322,7 +7322,51 @@ But heed the warning: [Avoid "naked" `union`s](#Ru-naked)
...
+            }
+        }
+    
+        ~Immutable_string()
+        {
+            if (size >= buffer_size)
+            {
+                delete string_ptr;
+            }
+        }
```
Comment (gdr-at-ms) [inline CppCoreGuidelines.md]: Style nit: Branches of if-statements that are single statements go by themselves, no need to package them in braces.

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -7322,7 +7322,51 @@ But heed the warning: [Avoid "naked" `union`s](#Ru-naked)
 
 ##### Example
 
-    ??? short-string optimization; safe union without dscriminant ???
+    // Short string optimization
+    
+    constexpr size_t buffer_size = 16;
+
```
Comment (gdr-at-ms) [inline CppCoreGuidelines.md]: Comments indicating why this value.

**Comment by gdr-at-ms** [comment]
Could you squash your commits?.

**Comment by Eliyahu-Ravuna** [comment]
Squashed.

File: CppCoreGuidelines.md [hunk 3]
```diff
@@ -7322,7 +7322,49 @@ But heed the warning: [Avoid "naked" `union`s](#Ru-naked)
...
+    constexpr size_t buffer_size = 16; // Slightly larger than the size of a pointer
+    
+    class Immutable_string
+    {
+    public:
+        Immutable_string(const char* str) :
+            size(strlen(str))
+        {
+            if (size < buffer_size)
+            {
```
Comment (gdr-at-ms) [inline CppCoreGuidelines.md]: You still have this instance though...

**Comment by Eliyahu-Ravuna** [comment]
All braces revised (including the braces of the union) to make them consistent with the rest of the examples.

**Pull request merged**